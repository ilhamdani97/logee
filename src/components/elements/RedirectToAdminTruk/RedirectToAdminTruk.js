import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { RedirectModal, useCookie } from 'logeeport-ui';
import { LOGEETRANS_URL } from '../../../services/order';
import jwt from 'jwt-simple';
import styles from './styles.scoped.css';


const RedirectToAdminTruk = ({ open, onClose,data, tipe }) => {

  useEffect(() => {
    const payload = useCookie.getCookie('userLogin') ? JSON.parse(useCookie.getCookie('userLogin')):null;
    const token = useCookie.getCookie('accessToken') || null;
    const ekripPayload = { ...payload,accessToken:token };
    const secret = 'xxx';
    const LogeeToken = jwt.encode(ekripPayload, secret);


    let logeeUrl = {
      'invoice':`${LOGEETRANS_URL}/invoice?type=single
      &order=${data.orderNumber}
      &id=${data.deliveryDetailId}
      &logeeport=${LogeeToken}`,
      'order':`${LOGEETRANS_URL}/pemesanan?id=${data.deliveryDetailId}&logeeport=${LogeeToken}`
    };


    if(open){
      setTimeout(()=>{
        window.open(logeeUrl[tipe],'_blank');
        onClose();
      },3000);
    }

  }, [open]);

  return (
    <RedirectModal open={open}>
      <p className={styles.headLoading}>Mengarahkan ke Logee Truk</p>
      <p className={styles.captionLoading}>
          Anda dapat melihat detail pemesanan truk lebih detail di Logee Truck.
      </p>
    </RedirectModal>
  );
};
RedirectToAdminTruk.defaultProps = {
  data:null,
  onClose:()=>{},
  open:false,
  tipe: ''

};

RedirectToAdminTruk.propTypes = {
  data:PropTypes.object,
  onClose: PropTypes.func,
  open: PropTypes.bool,
  tipe: PropTypes.string,
};
export default RedirectToAdminTruk;
