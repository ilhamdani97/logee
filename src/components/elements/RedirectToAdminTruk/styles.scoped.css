.root {
  background-color: var(--background-color-2);
  border-radius: 4px;
  width: 100%;
  box-sizing: border-box;
  display: block;
}
.noPadSide {
  padding: 8px 0;
}

.title {
  border-bottom: 2px solid var(--grey-color-3);
  padding: 12px;
  font-size: 18px;
  color: #000000;
  text-transform: capitalize;
  font-weight: 700;
}

.headLoading {
  font-size: 18px;
  font-weight: 600;
  margin-bottom: 16px;
  color: #212121;
  text-align: center;
}

.captionLoading {
  font-size: 14px;
  margin-bottom: 16px;
  color: #757575;
  text-align: center;
}
