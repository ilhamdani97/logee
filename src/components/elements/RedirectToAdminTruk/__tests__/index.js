import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import RedirectToAdminTruk from '../RedirectToAdminTruk';


const handleClose = jest.fn();
const data = {
  deliveryDetailId:'24324324',
  orderNumber:'12121'
};

describe('src/components/elements/RedirectToAdminTruk', () => {
  test('render', () => {

    const shallow = new ShallowRenderer();
    const tree = shallow.render(<RedirectToAdminTruk data={data} onClose={handleClose} open={true} tipe="order" />);
    expect(tree).toMatchSnapshot();
  });
});
