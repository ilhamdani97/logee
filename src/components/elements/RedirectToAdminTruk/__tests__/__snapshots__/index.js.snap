// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/RedirectToAdminTruk render 1`] = `
<RedirectModal
  className=""
  open={true}
>
  <p>
    Mengarahkan ke Logee Truk
  </p>
  <p>
    Anda dapat melihat detail pemesanan truk lebih detail di Logee Truck.
  </p>
</RedirectModal>
`;
